lib/rightscript_sync.rb

```diff
@@ -69,45 +69,66 @@ def set_metadata(*files)
   end
 
   desc "download <file> [-n -i] <script dash name>", "Download RightScript to file"
-  method_option :id, 
+  method_option :id,
     :desc => "RightScript ID number to download",
     :aliases => "-i",
     :type => :numeric
-  method_option :name, 
+  method_option :name,
     :desc => "RightScript Name to download",
     :aliases => "-n"
   def download(file)
-    raise ArgumentError.new("Either name or id must be supplied") unless options[:name] || options[:id]
-    raise ArgumentError.new("Filename to download to must be supplied") unless file && !file.empty?
-
-    if options[:id]
-      rightscript = api_client.right_scripts(:id => options[:id]).show
-    elsif options[:name]
-      rightscripts = api_client.right_scripts.index(:filter => ["name==#{options[:name]}"])
-      rightscripts = rightscripts.select { |rs| rs.revision.to_i == 0 && rs.name == options[:name] }
-      if rightscripts.length > 1
-        raise ArgumentError.new("More than one rightscript round with name #{options[:name]}")
-      elsif rightscripts.length == 0
-        raise ArgumentError.new("Could not find rightscript named #{options[:name]}")
+    if options[:name] || options[:id]
+      raise ArgumentError.new("Filename to download to must be supplied") unless file && !file.empty?
+
+      if options[:id]
+        rightscript = api_client.right_scripts(:id => options[:id]).show
+      elsif options[:name]
+        rightscript = get_rightscript_object(options[:name])
+        raise ArgumentError.new("Could not find rightscript named #{script_name}") if rightscript.nil?
```

Comment:
Actually, if you've entered this block both name and id were supplied, so it is assumed you are referring to a rightscript directly and not a folder/directory.  The other cases where name or id are missing are handled by the other paths.